### PR TITLE
Enable Access log

### PR DESCRIPTION
**Pull request by pilhuhn**
I had to provide a dummy principal for the unauthenticated case as otherwise creation of the access log fails with a NPE
(it is trying to get the name from the principal)

### CODE REVIEW

File: src/test/resources/application.properties [hunk 1]
```diff
@@ -10,8 +10,8 @@ quarkus.datasource.jdbc.url=jdbc:postgresql://192.168.1.139:5432/notifications-t
 # Flyway minimal config properties
 quarkus.flyway.migrate-at-start=true
 
-#quarkus.http.access-log.enabled=true
-quarkus.http.access-log.category=info
+quarkus.http.access-log.enabled=true
```
Comment (burmanm) [inline src/test/resources/application.properties:13]: I disabled this one on purpose, it simply spams the test log for no good reason.

**Comment by burmanm** [comment]
I think that NPE should be logged as a bug to the Quarkus team as that identityBuilder should work.

**Comment by pilhuhn** [comment]
On 21 Oct 2020, at 15:30, Michael Burman wrote:

> +quarkus.http.access-log.enabled=true
>
> I disabled this one on purpose, it simply spams the test log for no 
> good reason.

Leaving it disabled would not have found the NPE in tests, but in prod.

**Comment by burmanm** [comment]
I think it should stay disabled if it doesn't work reliably. What if Rbac returns a null response? Will it crash?

**Comment by pilhuhn** [comment]
If rbac returns null, then a lot of `principal.getAccount()` will NPE, so it doesn't matter here ;-)

**Comment by burmanm** [comment]
Principal shouldn't return null as that's configured from the input x-rh-identity header. If rbac returns null then that request will be thrown out and the logging shouldn't crash in that situation. Rbac authorization is not responsible for Principal item, authentication builds that one.

**Comment by pilhuhn** [comment]
That assumes that the x-rh-identity header is provided. The current code bails out early when the header is null and then the principal is null and principal.getAccount will NPE

**Comment by burmanm** [comment]
If x-rh-identity is missing, then no getAccount is ever called as AuthenticationException is thrown and the processing stops there.  It returns 403 correctly. In which situation would it get a NPE?

**Comment by pilhuhn** [comment]
No longer needed

**Pull request closed without merging**